src/rule_definition_tools.jl

```diff
@@ -140,7 +140,15 @@ function rule_from_partials(input_arg, ∂s...)
         conjugate_rule = :(Rule($Δs_tuple -> add($(∂_mul_Δs_conjugate...))))
         return quote
             $(∂_wirtinger_defs...)
-            WirtingerRule(typeof($input_arg), $primal_rule, $conjugate_rule)
+            let
```

Comment:
This behaves differently to my implementation if only some of the arguments are complex. In these cases not all arguments need to have Wirtinger derivatives, so we can still apply the fall-through on all the real arguments. Your approach is probably fine for a lot of functions, which use the promotion mechanism anyways, but could be unexpected in some cases.